lib/jekyll-linkpreview.rb

```diff
@@ -220,7 +222,9 @@ def render_linkpreview(properties)
 
       private
       def get_custom_template_path(properties)
-        File.join Dir.pwd, @@template_dir, properties.template_file
+        @source_dir.nil? || @source_dir.empty? ?
+          File.join(Dir.pwd, @@template_dir, properties.template_file) :
+          File.join(Dir.pwd, @source_dir, @@template_dir, properties.template_file)
```

Comment:
If we add `context` argument to this function, it should look like this. It would look way simpler IMO.

```ruby
private
def get_custom_template_path(context, properties)
  source_dir = get_source_dir_from(context)
  File.join Dir.pwd, source_dir, @@template_dir, properties.template_file

private
def get_source_dir_from(context)
  source_dir = context.registers[:site].config['source']
  source_dir.nil? ? '.' : source_dir
end
```

Added `get_source_dir_from`  just for readabilility. This is optional.

Reply:
Fixed as requested.  
Note, after I sent this PR I found out that if the site's source is not specified, then it will use `Dir.pwd` and mangle everything!  
Now that we know any valid path (both absolute and relative) can appear as `source_dir`, I filtered the configuration value using `File.absolute_path config_val, Dir.pwd` so that we can get the expected result for any cases.  
There's one concern: I put `Dir.pwd` as second argument for clarification purposes, but we can omit it and still observe the same behavior (cf.[`File.absolute_path(file_name, dir_string=nil)`](https://docs.ruby-lang.org/ja/latest/method/File/s/absolute_path.html)). Should I drop it?

Reply:
> Should I drop it?

Either is fine with me. We can just keep it.

But when `source_dir.nil?` is true, this function returns '.' which is not an absolute path.
You should fix it like this:

```
        source_dir = context.registers[:site].config['source']
        File.absolute_path source_dir.nil? ? '.' : source_dir, Dir.pwd
```

Reply:
As I wrote here: https://github.com/ysk24ok/jekyll-linkpreview/pull/32#discussion_r1038746271, I reckon `context.registers[:site].config['source']` is a guaranteed non-nil.

Reply:
Then, why not delete the if branch?

```rb
      private
      def get_source_dir_from(context)
        File.absolute_path context.registers[:site].config['source'], Dir.pwd
      end
```

Reply:
You're right, I totally overlooked that. I've updated the diff.